### PR TITLE
Nh3 support

### PR DESCRIPTION
**Pull request by roberbike**
Resuelto conflictos con BME 680 , inicializacion y lectura de sensores NH3 y CO correcta.

### CODE REVIEW

**Comment by roberbike** [comment]
20:08:58.980 >
20:08:58.980 > rst:0x1 (POWERON_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
20:08:58.980 > configsip: 0, SPIWP:0xee
20:08:58.980 > clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
20:08:58.995 > mode:DIO, clock div:2
20:08:58.999 > load:0x3fff0030,len:1184
20:08:59.001 > load:0x40078000,len:13104
20:08:59.004 > load:0x40080400,len:3036
20:08:59.006 > entry 0x400805e4
20:08:59.407 > 
20:08:59.407 > == Sensor test setup ==
20:08:59.409 >
20:08:59.409 > -->[SETUP] Detecting sensors..
20:08:59.412 > -->[SLIB] CanAirIO SensorsLib    : v0.6.7r374
20:08:59.416 > -->[SLIB] sensorslib devmod      : true      
20:08:59.419 > -->[SLIB] temperature offset     : 0.00      
20:08:59.422 > -->[SLIB] altitude offset        : 0.00      
20:08:59.425 > -->[SLIB] sea level pressure     : 1013.25 hPa     
20:08:59.431 > -->[SLIB] only i2c sensors       : false
20:08:59.432 > -->[SLIB] UART detecting type    : Auto
20:08:59.436 > -->[SLIB] UART init with speed   : 9600 TX:16 RX:17
20:08:59.440 > -->[SLIB] UART COMM port         : Serial2
20:09:00.456 > -->[SLIB] UART detection msg     : no data 
20:09:01.463 > -->[SLIB] UART detection msg     : no data        
20:09:01.467 > [W][SLIB] GCJA5 UART msg         : invalid header 
20:09:02.470 > -->[SLIB] UART detection msg     : no data 
20:09:03.477 > -->[SLIB] UART detection msg     : no data        
20:09:03.481 > [W][SLIB] GCJA5 UART msg         : invalid header 
20:09:04.480 > -->[SLIB] UART detection msg     : no data 
20:09:05.491 > -->[SLIB] UART detection msg     : no data        
20:09:05.495 > [W][SLIB] GCJA5 UART msg         : invalid header 
20:09:05.498 > -->[SLIB] UART sensors detected  : 0
20:09:05.502 > -->[SLIB] attempt enable sensor  : SCD30
20:09:05.505 > -->[SLIB] attempt enable sensor  : SPS30
20:09:05.509 > -->[SLIB] SPS30 Serial number    :  E8367136CBEDA5BB
20:09:05.513 > -->[SLIB] SPS30 product name     :  00080000        
20:09:05.517 > -->[SLIB] SPS30 firmware level   :  2.3
20:09:05.520 > -->[SLIB] SPS30 Hardware level   : 0
20:09:05.524 > -->[SLIB] SPS30 SHDLC protocol   : 0.0
20:09:05.527 > -->[SLIB] SPS30 Library level    : 1.4
20:09:07.514 > -->[SLIB] SPS30 Detected SPS30 via I2C. 
20:09:08.523 > -->[SLIB] SPS30 Measurement OK
20:09:08.526 > -->[SLIB] sensor registered      : SPS30         :D
20:09:08.530 > -->[SLIB] attempt enable sensor  : GCJA5
20:09:08.533 > -->[SLIB] attempt enable sensor  : SCD4X
20:09:09.031 > [W][SLIB] SCD4x stopping error   : 267   
20:09:09.031 > -->[SLIB] attempt enable sensor  : BMP280
20:09:09.031 > -->[SLIB] attempt enable sensor  : BME280
20:09:09.031 > -->[SLIB] attempt enable sensor  : BME680
20:09:09.049 > -->[SLIB] attempt enable sensor  : AM232X
20:09:09.052 > -->[SLIB] attempt enable sensor  : SHT31
20:09:09.056 > -->[SLIB] attempt enable sensor  : AHTXX
20:09:09.149 > -->[SLIB] attempt enable sensor  : DFRCO
20:09:11.354 > -->[SLIB] sensor registered      : DFRCO         :D
20:09:11.358 > -->[SLIB] attempt enable sensor  : DFRNH3
20:09:13.558 > -->[SLIB] sensor registered      : DFRNH3        :D
20:09:13.562 > -->[SLIB] Sensors count          : 3 (SPS30,DFRCO,DFRNH3,)
20:09:15.604 > -->[SLIB] SPS30 read             : done! 
20:09:18.061 > -->[SLIB] Sensors values         : PM1:6.0 PM2.5:7.0 PM4:7.0 PM10:7.0 CO:166.0 NH3:0.9 
20:09:18.069 > -->[SLIB] Sensors count          : 3 (SPS30,DFRCO,DFRNH3,)
20:09:18.074 > -->[SLIB] Sensors units count    : 6 (PM1,PM2.5,PM4,PM10,CO,NH3,)
20:09:18.079 > ======= E X A M P L E   T E S T =========
20:09:18.083 > -->[MAIN] Sensors detected count : 3
20:09:18.086 > -->[MAIN] Sensors units count    : 6
20:09:18.089 > -->[MAIN] Sensors devices names  : SPS30,DFRCO,DFRNH3,
20:09:18.094 > -->[MAIN] Preview sensor values:
20:09:18.097 > -->[MAIN] PM1:   6.0     ug/m3
20:09:18.100 > -->[MAIN] PM2.5: 7.0     ug/m3
20:09:18.100 > -->[MAIN] PM4:   7.0     ug/m3
20:09:18.104 > -->[MAIN] PM10:  7.0     ug/m3
20:09:18.106 > -->[MAIN] CO:    166.0   ppm
20:09:18.108 > -->[MAIN] NH3:   0.9     ppm
20:09:25.605 > -->[SLIB] SPS30 read             : done! 
20:09:28.049 > -->[SLIB] Sensors values         : PM1:6.0 PM2.5:6.0 PM4:6.0 PM10:6.0 CO:166.0 NH3:0.9 
20:09:28.064 > -->[SLIB] Sensors count          : 3 (SPS30,DFRCO,DFRNH3,)
20:09:28.064 > -->[SLIB] Sensors units count    : 6 (PM1,PM2.5,PM4,PM10,CO,NH3,)
20:09:28.080 > ======= E X A M P L E   T E S T =========
20:09:28.084 > -->[MAIN] Sensors detected count : 3
20:09:28.087 > -->[MAIN] Sensors units count    : 6
20:09:28.091 > -->[MAIN] Sensors devices names  : SPS30,DFRCO,DFRNH3,
20:09:28.095 > -->[MAIN] Preview sensor values:
20:09:28.098 > -->[MAIN] PM1:   6.0     ug/m3
20:09:28.101 > -->[MAIN] PM2.5: 6.0     ug/m3
20:09:28.103 > -->[MAIN] PM4:   6.0     ug/m3
20:09:28.105 > -->[MAIN] PM10:  6.0     ug/m3
20:09:28.107 > -->[MAIN] CO:    166.0   ppm
20:09:28.109 > -->[MAIN] NH3:   0.9     ppm
20:09:35.605 > -->[SLIB] SPS30 read             : done! 
20:09:38.063 > -->[SLIB] Sensors values         : PM1:5.0 PM2.5:5.0 PM4:5.0 PM10:5.0 CO:166.0 NH3:0.9 
20:09:38.071 > -->[SLIB] Sensors count          : 3 (SPS30,DFRCO,DFRNH3,)
20:09:38.075 > -->[SLIB] Sensors units count    : 6 (PM1,PM2.5,PM4,PM10,CO,NH3,)
20:09:38.081 > ======= E X A M P L E   T E S T =========
20:09:38.084 > -->[MAIN] Sensors detected count : 3
20:09:38.088 > -->[MAIN] Sensors units count    : 6
20:09:38.091 > -->[MAIN] Sensors devices names  : SPS30,DFRCO,DFRNH3,
20:09:38.096 > -->[MAIN] Preview sensor values:
20:09:38.099 > -->[MAIN] PM1:   5.0     ug/m3
20:09:38.101 > -->[MAIN] PM2.5: 5.0     ug/m3
20:09:38.103 > -->[MAIN] PM4:   5.0     ug/m3
20:09:38.105 > -->[MAIN] PM10:  5.0     ug/m3
20:09:38.108 > -->[MAIN] CO:    166.0   ppm
20:09:38.110 > -->[MAIN] NH3:   0.9     ppm
20:09:45.605 > -->[SLIB] SPS30 read             : done! 
20:09:48.063 > -->[SLIB] Sensors values         : PM1:5.0 PM2.5:5.0 PM4:5.0 PM10:5.0 CO:165.0 NH3:0.9 
20:09:48.070 > -->[SLIB] Sensors count          : 3 (SPS30,DFRCO,DFRNH3,)
20:09:48.075 > -->[SLIB] Sensors units count    : 6 (PM1,PM2.5,PM4,PM10,CO,NH3,)
20:09:48.080 > ======= E X A M P L E   T E S T =========
20:09:48.085 > -->[MAIN] Sensors detected count : 3
20:09:48.087 > -->[MAIN] Sensors units count    : 6
20:09:48.090 > -->[MAIN] Sensors devices names  : SPS30,DFRCO,DFRNH3,
20:09:48.096 > -->[MAIN] Preview sensor values:
20:09:48.099 > -->[MAIN] PM1:   5.0     ug/m3
20:09:48.101 > -->[MAIN] PM2.5: 5.0     ug/m3
20:09:48.104 > -->[MAIN] PM4:   5.0     ug/m3
20:09:48.106 > -->[MAIN] PM10:  5.0     ug/m3
20:09:48.108 > -->[MAIN] CO:    165.0   ppm
20:09:48.110 > -->[MAIN] NH3:   0.9     ppm
20:09:55.606 > -->[SLIB] SPS30 read             : done! 
20:09:58.064 > -->[SLIB] Sensors values         : PM1:5.0 PM2.5:5.0 PM4:5.0 PM10:5.0 CO:164.0 NH3:0.9 
20:09:58.071 > -->[SLIB] Sensors count          : 3 (SPS30,DFRCO,DFRNH3,)
20:09:58.076 > -->[SLIB] Sensors units count    : 6 (PM1,PM2.5,PM4,PM10,CO,NH3,)
20:09:58.081 > ======= E X A M P L E   T E S T =========
20:09:58.085 > -->[MAIN] Sensors detected count : 3
20:09:58.089 > -->[MAIN] Sensors units count    : 6
20:09:58.093 > -->[MAIN] Sensors devices names  : SPS30,DFRCO,DFRNH3,
20:09:58.097 > -->[MAIN] Preview sensor values:
20:09:58.099 > -->[MAIN] PM1:   5.0     ug/m3
20:09:58.102 > -->[MAIN] PM2.5: 5.0     ug/m3
20:09:58.104 > -->[MAIN] PM4:   5.0     ug/m3
20:09:58.107 > -->[MAIN] PM10:  5.0     ug/m3
20:09:58.108 > -->[MAIN] CO:    164.0   ppm
20:09:58.110 > -->[MAIN] NH3:   0.9     ppm
20:10:05.605 > -->[SLIB] SPS30 read             : done! 
20:10:08.064 > -->[SLIB] Sensors values         : PM1:5.0 PM2.5:5.0 PM4:5.0 PM10:5.0 CO:165.0 NH3:0.9 
20:10:08.072 > -->[SLIB] Sensors count          : 3 (SPS30,DFRCO,DFRNH3,)
20:10:08.076 > -->[SLIB] Sensors units count    : 6 (PM1,PM2.5,PM4,PM10,CO,NH3,)
20:10:08.082 > ======= E X A M P L E   T E S T =========
20:10:08.085 > -->[MAIN] Sensors detected count : 3
20:10:08.089 > -->[MAIN] Sensors units count    : 6
20:10:08.092 > -->[MAIN] Sensors devices names  : SPS30,DFRCO,DFRNH3,
20:10:08.097 > -->[MAIN] Preview sensor values:
20:10:08.100 > -->[MAIN] PM1:   5.0     ug/m3
20:10:08.102 > -->[MAIN] PM2.5: 5.0     ug/m3
20:10:08.104 > -->[MAIN] PM4:   5.0     ug/m3
20:10:08.106 > -->[MAIN] PM10:  5.0     ug/m3
20:10:08.109 > -->[MAIN] CO:    165.0   ppm
20:10:08.111 > -->[MAIN] NH3:   0.9     ppm
20:10:15.606 > -->[SLIB] SPS30 read             : done! 
20:10:18.064 > -->[SLIB] Sensors values         : PM1:4.0 PM2.5:5.0 PM4:5.0 PM10:5.0 CO:164.0 NH3:0.9 
20:10:18.072 > -->[SLIB] Sensors count          : 3 (SPS30,DFRCO,DFRNH3,)
20:10:18.077 > -->[SLIB] Sensors units count    : 6 (PM1,PM2.5,PM4,PM10,CO,NH3,)
20:10:18.082 > ======= E X A M P L E   T E S T =========
20:10:18.086 > -->[MAIN] Sensors detected count : 3
20:10:18.089 > -->[MAIN] Sensors units count    : 6
20:10:18.092 > -->[MAIN] Sensors devices names  : SPS30,DFRCO,DFRNH3,
20:10:18.097 > -->[MAIN] Preview sensor values:
20:10:18.100 > -->[MAIN] PM1:   4.0     ug/m3
20:10:18.102 > -->[MAIN] PM2.5: 5.0     ug/m3
20:10:18.105 > -->[MAIN] PM4:   5.0     ug/m3
20:10:18.107 > -->[MAIN] PM10:  5.0     ug/m3
20:10:18.109 > -->[MAIN] CO:    164.0   ppm
20:10:18.111 > -->[MAIN] NH3:   0.9     ppm
20:10:25.607 > -->[SLIB] SPS30 read             : done! 
20:10:28.065 > -->[SLIB] Sensors values         : PM1:4.0 PM2.5:4.0 PM4:4.0 PM10:4.0 CO:164.0 NH3:0.9 
20:10:28.072 > -->[SLIB] Sensors count          : 3 (SPS30,DFRCO,DFRNH3,)
20:10:28.077 > -->[SLIB] Sensors units count    : 6 (PM1,PM2.5,PM4,PM10,CO,NH3,)
20:10:28.082 > ======= E X A M P L E   T E S T =========
20:10:28.086 > -->[MAIN] Sensors detected count : 3
20:10:28.089 > -->[MAIN] Sensors units count    : 6
20:10:28.093 > -->[MAIN] Sensors devices names  : SPS30,DFRCO,DFRNH3,
20:10:28.097 > -->[MAIN] Preview sensor values:
20:10:28.100 > -->[MAIN] PM1:   4.0     ug/m3
20:10:28.103 > -->[MAIN] PM2.5: 4.0     ug/m3
20:10:28.105 > -->[MAIN] PM4:   4.0     ug/m3
20:10:28.107 > -->[MAIN] PM10:  4.0     ug/m3
20:10:28.109 > -->[MAIN] CO:    164.0   ppm
20:10:28.111 > -->[MAIN] NH3:   0.9     ppm
20:10:35.607 > -->[SLIB] SPS30 read             : done! 
20:10:38.065 > -->[SLIB] Sensors values         : PM1:4.0 PM2.5:4.0 PM4:4.0 PM10:4.0 CO:164.0 NH3:0.9 
20:10:38.073 > -->[SLIB] Sensors count          : 3 (SPS30,DFRCO,DFRNH3,)
20:10:38.077 > -->[SLIB] Sensors units count    : 6 (PM1,PM2.5,PM4,PM10,CO,NH3,)
20:10:38.083 > ======= E X A M P L E   T E S T =========
20:10:38.086 > -->[MAIN] Sensors detected count : 3
20:10:38.090 > -->[MAIN] Sensors units count    : 6
20:10:38.093 > -->[MAIN] Sensors devices names  : SPS30,DFRCO,DFRNH3,
20:10:38.098 > -->[MAIN] Preview sensor values:
20:10:38.101 > -->[MAIN] PM1:   4.0     ug/m3
20:10:38.104 > -->[MAIN] PM2.5: 4.0     ug/m3
20:10:38.105 > -->[MAIN] PM4:   4.0     ug/m3
20:10:38.107 > -->[MAIN] PM10:  4.0     ug/m3
20:10:38.110 > -->[MAIN] CO:    164.0   ppm
20:10:38.112 > -->[MAIN] NH3:   0.9     ppm
20:10:45.607 > -->[SLIB] SPS30 read             : done! 
20:10:48.065 > -->[SLIB] Sensors values         : PM1:4.0 PM2.5:5.0 PM4:5.0 PM10:5.0 CO:164.0 NH3:0.9 
20:10:48.072 > -->[SLIB] Sensors count          : 3 (SPS30,DFRCO,DFRNH3,)
20:10:48.077 > -->[SLIB] Sensors units count    : 6 (PM1,PM2.5,PM4,PM10,CO,NH3,)

File: src/Sensors.cpp [hunk 1]
```diff
@@ -993,13 +993,15 @@ void Sensors::DFRobotNH3Read() {
     String gastype = dfrNH3.queryGasType();
     nh3 = dfrNH3.readGasConcentrationPPM();
     unitRegister(UNIT::NH3);
+   
 }
 
 void Sensors::DFRobotCORead() {
     if (!dfrCO.begin()) return;
     String gastype = dfrCO.queryGasType();
```
Comment (hpsaturn) [inline src/Sensors.cpp:1001]: if we don't need it, please remove this line. Please confirm with the device.

File: src/Sensors.cpp [hunk 2]
```diff
@@ -1582,23 +1584,25 @@ void Sensors::DFRobotCOInit() {
...
 
 void Sensors::DFRobotNH3Init() {
   sensorAnnounce(SENSORS::SDFRNH3);
-  dfrNH3 = DFRobot_GAS_I2C(&Wire, 0x77);
+  dfrNH3 = DFRobot_GAS_I2C(&Wire, 0x76); // 0x77 y 0x75 used by bme680
   if (!dfrNH3.begin()) return;
   //Mode of obtaining data: the main controller needs to request the sensor for data
-//   dfrNH3.changeAcquireMode(dfrNH3.PASSIVITY);
+  dfrNH3.changeAcquireMode(dfrNH3.PASSIVITY);
   delay(1000);  // TODO: we need validate if we need that
```
Comment (hpsaturn) [inline src/Sensors.cpp:1602]: we need this big delays? please try to test if it is not necessary

Comment (hpsaturn) [inline src/Sensors.cpp:1602]: we need this big delays? please try to test if it is not necessary

File: src/Sensors.cpp [hunk 3]
```diff
@@ -1582,23 +1584,25 @@ void Sensors::DFRobotCOInit() {
...
-
   //Turn on temperature compensation: gas.ON : turn on
   dfrNH3.setTempCompensation(dfrNH3.ON);
+  delay(1000);
```
Comment (hpsaturn) [inline src/Sensors.cpp:1605]: we need this big delays? please try to test if it is not necessary

**Review by hpsaturn — CHANGES_REQUESTED** [review]

**Review by hpsaturn — APPROVED** [review]

**Pull request merged**